src/pudl/helpers.py

```diff
@@ -36,6 +37,54 @@
 tz_finder = timezonefinder.TimezoneFinder()
 
 
+def add_fips_ids(df, state_col="state", county_col="county", vintage=2015):
+    """Add State and County FIPS IDs to a dataframe."""
+    af = addfips.AddFIPS(vintage=vintage)
+    # Lookup the state and county FIPS IDs and add them to the dataframe:
+    df["state_id_fips"] = df.apply(
+        lambda x: af.get_state_fips(state=x.state), axis=1)
+    logger.info(
+        f"Assigned state FIPS codes for "
+        f"{len(df[df.state_id_fips.notnull()])/len(df):.2%} of records."
+    )
+    df["county_id_fips"] = df.apply(
+        lambda x: af.get_county_fips(state=x.state, county=x.county), axis=1)
+    df["county_id_fips"] = df.county_id_fips.fillna(pd.NA)
+    logger.info(
+        f"Assigned county FIPS codes for "
+        f"{len(df[df.county_id_fips.notnull()])/len(df):.2%} of records."
+    )
+    return df
+
+
+def clean_eia_counties(df, fixes, state_col="state", county_col="county"):
+    """Replace non-standard county names with county nmes from US Census."""
+    df = df.copy()
+    df[county_col] = (
+        df[county_col].str.strip()
+        .str.replace(r"\s+", " ")  # Condense multiple whitespace chars.
```

Comment:
I'm wondering if we could compile these required changes in a dictionary or something of the sort so the specific stringy logic isn't stored within the function.

Reply:
Maybe these functions shouldn't be in helpers -- they're really more specific than that, and only intended to help out in the transformation of EIA tables that contain lists of county names, which seem to share the same set of naming issues (at least between EIA 860 and EIA 861). Would this level of specificity be more acceptable if it were in the `pudl.transform.eia` module?

Reply:
I don't mind it in helpers, but i can see it being a mostly transform only function. and seemingly only eia right now. But what I was attempting to note is that maybe the specifics of what needs to be replaced could be stored as a little dictionary as a global variable in whatever module it lives in. But it would also be nice if we could overwrite (or add to) that dictionary if we were using this on a different set of messy county names.

Reply:
also this is fine the way it is! this is not a blocking comment - it was just something that might be nice.

Reply:
Okay I think I'm going to leave this as is for the moment, just because the function has a very narrow applicability right now (fixing counties in the EIA 860 Plants table, and also in the EIA 861 Service Territory table), but I think this is another example where it might be good to have a more general infrastructure for applying "fixes" to a dataframe, and separating the content of the fixes from the infrastructure that applies them. So, somehow specifying a mask that selects records, a column within those records that will be assigned, and the value to which it should be assigned.  I feel like this has to be something that already exists in some form, since it seems like a very general problem in data wrangling?

Reply:
yes, exactly. and i think leaving it for now makes sense!